torchvision/transforms/functional_tensor.py

```diff
@@ -236,3 +276,51 @@ def _blend(img1, img2, ratio):
     # type: (Tensor, Tensor, float) -> Tensor
     bound = 1 if img1.dtype in [torch.half, torch.float32, torch.float64] else 255
     return (ratio * img1 + (1 - ratio) * img2).clamp(0, bound).to(img1.dtype)
+
+
+def _rgb2hsv(img):
+    r, g, b = img[0], img[1], img[2]
+
+    maxc, _ = torch.max(img, dim=0)
+    minc, _ = torch.min(img, dim=0)
+    uv = maxc
+
+    cr = maxc.to(dtype=torch.float32) - minc.to(dtype=torch.float32)
+    s = cr / maxc.to(dtype=torch.float32)
+    rc = (maxc-r).to(dtype=torch.float32) / cr
+    gc = (maxc-g).to(dtype=torch.float32) / cr
+    bc = (maxc-b).to(dtype=torch.float32) / cr
+
+    s = (maxc != minc) * s
+    hr = (maxc == r) * (bc - gc)
+    hg = ((maxc == g) & (maxc != r)) * (2.0 + rc - bc)
+    hb = ((maxc != g) & (maxc != r)) * (4.0 + gc - rc)
```

Comment:
I think if you keep the indices from
```python
maxc, max_idxs = torch.max(img, dim=0)
```
then you can avoid these set of equalities to find where the max came from. This part could even be batched and vectorized
```python
max_mask = max_idxs[..., None] == torch.tensor([0, 1, 2])
temp1 = (bc - gc)
temp2 = (2.0 + rc - bc)
temp3 = (4.0 + gc - rc)
temp = torch.stack([temp1, temp2, temp3], dim=-1)
h = (max_mask * temp).sum(-1)
```

Although having 2 additions might be more efficient than calling `sum`

Reply:
Also i have to ensure I dont add twice if the img has duplicate values. I dont see how the above code avoids that from happening. Please let me know.

Reply:
The main problem I was facing with this was when img has duplicate values and the max occurs at two or more locations. In that case there is an order that needs to be followed. For instance maxc == r > maxc ==g > maxc = b. And we need to add only one of the rc, gc, or bc to h.  I dont understand how we can enforce this order and mutually exclusive thing from your suggestion. Please let me know.

Reply:
Sounds good. There could be ways of working around this issue, but it would probably make the code less readable.